### PR TITLE
docs: refactoring and cleanup

### PR DESCRIPTION
**Pull request by dmke**
I just wanted to add a small section about being a good netizen (for #1656), but ended up splitting the CLI usage examples into easier digestible parts.

By doing so `usage/cli/examples/` would have been gone, and direct links (think: bookmarks) would yield a 404 Not Found error. Github Pages doesn't allow redirects (apart from ugly hacks involving `<meta http-equiv="refresh" content="0; URL=...">`), so I've kept the examples page as a landing page, but hidden from the main menu. Users following a direct link can use it reorient themselves.

<details><summary>Screenshot</summary>

image

</details>

I've also updated, changed, harmonized and reshuffled a lot of things:

- Markdown links now use [Hugo's cross reference helpers](https://gohugo.io/content-management/cross-references/) to resolve the target (`[text](usage/cli#port-usage)` → `[text]({{< ref "usage/cli" >}})`).
- `dns/softlayer/` was redundant and I have removed it. I assumed this was overseen when refactoring PR #1487.
- the CLI examples now consistently use long-form flags (`--email` instead of `-m`)
- example email addresses are now consistent (`you@example.com`)
- `usage/dns/manual/` looked borked, I've recreated the example
- I've added a link to Github discussions to the sidebar
- the main body width is now constricted to max. 72em. I'm using a high-resolution, wide-screen monitor and now the text is not thinned over the whole screen. For resolutions less than ~1400px not much will change

  <details><summary>Screenshot</summary>

  maxwidth

  </details>

- The generated DNS provider files have now some meta data in their frontmatter. Using this meta data, I have reformatted the provider listing on `dns/#dns-providers`:

  <details><summary>Screenshot</summary>

  image

  </details>
For easier comparison, I've uploaded the generated site to https://mm.dmke.org/lego/

### CODE REVIEW

**Comment by dmke** [comment]
I'm now satisfied with the result and I have updated the PR description.

I tried to keep the commits small, but changes on the `internal/dnsdocs/dns.md.tmpl` inevitably produced also changes on the `docs/content/dns/zz_*.md` (who would've thunk?)

File: docs/layouts/shortcodes/tableofdnsproviders.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          <td>
+            <a href="{{ .RelPermalink }}">{{ .Title }}</a>
+          </td>
+          <td>
+            {{- if $params.url -}}
+              <a href="{{ $params.url }}" target="_blank">Website</a>
+            {{- end -}}
+          </td>
+          <td>
+            <kbd>{{ $params.code }}</kbd>
```
Comment (ldez) [inline docs/layouts/shortcodes/tableofdnsproviders.html:25]:
```suggestion
            <code>{{ $params.code }}</code>
```

File: internal/dnsdocs/dns.md.tmpl [hunk 2]
```diff
@@ -18,7 +21,9 @@ Configuration for [{{ .Name }}]({{ .URL }}).
 
 <!--more-->
 
-- Code: `{{ .Code }}`
+- Code: `--dns {{ .Code }}`
```
Comment (ldez) [inline internal/dnsdocs/dns.md.tmpl:24]:
```suggestion
- Code: `{{ .Code }}`
```

**Review by ldez — CHANGES_REQUESTED** [review]

**Review by ldez — APPROVED** [review]
Great job, the doc lacked a bit of love, it's great that you took the time to improve it.

**Pull request merged**